### PR TITLE
Add few Interop Marshal APIs

### PR DESCRIPTION
**Pull request by shrah**
Add the following interop Marshal APIs:

M:System.Runtime.InteropServices.Marshal.GenerateGuidForType(System.Type)
M:System.Runtime.InteropServices.Marshal.GenerateProgIdForType(System.Type)
M:System.Runtime.InteropServices.Marshal.GetComObjectData(System.Object,System.Object)
M:System.Runtime.InteropServices.Marshal.GetHINSTANCE(System.Reflection.Module)
M:System.Runtime.InteropServices.Marshal.GetIDispatchForObject(System.Object)
M:System.Runtime.InteropServices.Marshal.GetTypedObjectForIUnknown(System.IntPtr,System.Type)
M:System.Runtime.InteropServices.Marshal.PtrToStringAuto(System.IntPtr)
M:System.Runtime.InteropServices.Marshal.SetComObjectData(System.Object,System.Object,System.Object)
M:System.Runtime.InteropServices.Marshal.StringToCoTaskMemAuto(System.String)
M:System.Runtime.InteropServices.Marshal.StringToHGlobalAuto(System.String)

### CODE REVIEW

**Comment by shrah** [comment]
@yizhang82 @tijoytom  Please take a look.

File: src/vm/ecalllist.h [hunk 1]
```diff
@@ -1603,10 +1603,10 @@ FCFuncStart(gInteropMarshalFuncs)
     FCFuncElement("DestroyStructure", MarshalNative::DestroyStructure)
     FCFuncElement("UnsafeAddrOfPinnedArrayElement", MarshalNative::FCUnsafeAddrOfPinnedArrayElement)
     FCFuncElement("GetExceptionCode", ExceptionNative::GetExceptionCode)
+    QCFuncElement("GetHINSTANCE", COMModule::GetHINSTANCE)
```
Comment (yizhang82) [inline src/vm/ecalllist.h]: According to https://github.com/mono/mono/blob/master/mono/metadata/icall.c#L5739, mono returns -1 in non-Windows platforms (and I've confirmed that behavior). We should do the same. I took a quick look at our own code and I suspect we'll return the base address correctly in non-windows as well (but I haven't confirmed that theory - Faizur should be in a good place to test it). Assuming that's the case, we can special case it.

**Comment by yizhang82** [comment]
LGTM. Thanks, Faizur! 
/cc @alexperovich @weshaggard  @danmosemsft

File: src/vm/ecalllist.h (hunk 1 above)
Comment (shrah) [inline src/vm/ecalllist.h]: Yes mono returns -1. I have change it to ensure we return -1 for non-windows platform as well. I have updated the PR and squashed the changes.

**Pull request merged**